doc/source/release/2.0.1-notes.rst

```diff
@@ -5,7 +5,70 @@ NumPy 2.0.1 Release Notes
 ==========================
 
 NumPy 2.0.1 is a maintenance release that fixes bugs and regressions
-discovered after the 2.0.0 release.
+discovered after the 2.0.0 release. NumPy 2.0.1 is the last planned
+release in the 2.0.x series, 2.1.0rc1 should be out shortly.
 
 The Python versions supported by this release are 3.9-3.12.
 
+Improvements
+============
```

Comment:
Not sure sure where to comment this, but the text on the [release page](https://github.com/numpy/numpy/releases/tag/v2.0.1) shows:

> _**NOTE**_: Do not use the GitHub generated source* files, they are garbage.

Can you clarify a bit what's the issue?

PS. the `*` after "source" looks like a footnote, but there isn't one to be found - better to remove it perhaps.

Reply:
If you look at the `Assets` on the release page, you will see two files labeled `Source code`. They are automatically generated by GitHub and cannot be used to build NumPy. We would delete them if we could, but we can't, and occasionally someone downloads one, the build fails, and they open an issue. There was discussion of that last Wednesday, and the upshot was that we should warn not to use those files.

Reply:
And now is seems GitHub is looking at the notes and listing contributor names prefixed with '@` in a Contributors section. I wish they would stop trying to be helpful.

Reply:
Aside from missing submodules, what's the issue with building from source? I really wish there were no "invisible" source transformations that happen between the tag and the PyPI tarball, for example.

Reply:
IIRC, GitHub also did not pay attention to such things as the `MANIFEST.in` when we were using setup.py, nor handle the license files correctly. Of course there are transformations, files are compiled and libraries linked among other things, and the source distributions make sure the relevant files are included. If you want bare bones, install the repo.

Reply:
I understand all that, but saying the bare sources are "garbage" (as the release notes currently do) seems like FUD to me.

It could be formulated differently, e.g.: "If you want to build from source, we strongly recommend using `numpy-2.0.1.tar.gz`; we do not provide support for building from the archive that are autogenerated by Github (under "Source Code")."

> Of course there are transformations, files are compiled and libraries linked among other things, and the source distributions make sure the relevant files are included.

Nothing should be compiled into the sources tarball (e.g. running cython). That's the role of the build script (and the respective tool specification), not the tarball. The bigger the gap between the bare sources and the published tarball, the larger the surface for latent bugs (and security issues).

If it weren't for the convenience of not having to deal with the submodules, I'd already be using the github sources on the numpy-feedstock in conda-forge (as indeed we do for scipy). If however stuff is compiled and linked(?!) already in the sources, I'll probably have to bite that bullet though.

CC @rgommers, if you have thoughts on this.

Reply:
> It could be formulated differently, e.g.: "If you want to build from source, we strongly recommend using `numpy-2.0.1.tar.gz`; we do not provide support for building from the archive that are autogenerated by Github (under "Source Code")."

This sounds good to me. 

> Nothing should be compiled into the sources tarball (e.g. running cython).

That's not quite correct - the git archive metadata is generated (so we can have `np.version.git_revision`) and that cannot be avoided. The only supported methods of building from source are from a git repo clone or from an sdist. It is not possible to get an sdist from a "Source code" tarball.

Reply:
> The only supported methods of building from source are from a git repo clone or from an sdist.

Fine by me. OTOH I don't see a difference between a repo clone and the generated sources (with the only exception being submodules, because of `git archive`'s lack of support for that, and absence of a `.git` folder in the latter), so this was mainly about understanding what's meant by "garbage" (while being completely fine with not supporting those builds).

Reply:
The phrasing about "garbage" is still there, but FWIW, I've now [switched](https://github.com/conda-forge/numpy-feedstock/pull/322) conda-forge to building from raw sources (without any changes, except populating submodules). We need to do that anyway to be able to patch the vendored meson (c.f. #27044)